framework/src/base/Attributes.C

```diff
@@ -309,38 +311,72 @@ AttribPreIC::isEqual(const Attribute & other) const
   return isMatch(other);
 }
 
-void
-AttribPreAux::initFrom(const MooseObject * /*obj*/)
-{
-}
 bool
 AttribPreAux::isMatch(const Attribute & other) const
 {
   auto a = dynamic_cast<const AttribPreAux *>(&other);
-  return a && (a->_val == _val);
+
+  if (!a)
+    return false;
+
+  mooseAssert(a->_vals.size() == 1,
+              "Anticipated this function being passed an AttribPreAux object with a _vals member "
+              "containing only one item");
```

Comment:
I'm not sure this assert is quite right.  I think it would be safer to assert that neither `this->_vals` nor `a->_vals` are size zero.  Since it is possible that either one could have more than one entry - we want to preserve symmetry - i.e. `foo->isMatch(bar)` should return the same result as `bar->isMatch(foo)`.

Reply:
Let me explain why I currently have `a->_vals` checking that there is exactly one element.  I think this is still the correct way to do this, but let me know what you think.

`FEProblemBase::execute` calls `FEProblemBase::computeUserObjects` which then calls `FEProblemBase::computeUserObjectsInternal` and the first part of this routine is shown below. So the idea is that when `AttribSubdomains::isMatch` is called, it is passed an Attib object which was created in the way shown below, that is with exactly one exec flag. 

```c++
void
FEProblemBase::computeUserObjectsInternal(const ExecFlagType & type,
                                          const Moose::AuxGroup & group,
                                          TheWarehouse::Query & query)
{
  if (group == Moose::PRE_IC)
    query.condition<AttribPreIC>(true);
  else if (group == Moose::PRE_AUX)
    query.condition<AttribPreAux>(type);
  else if (group == Moose::POST_AUX)
    query.condition<AttribPostAux>(type);
```

Although with the code as it currently is, `a->_vals` would always only have one element, someone could change the code in the future such that the query attrib object is passed a std::set of exec flags instead of just one. The reason I want to trip an assertion in this case is that I'm not sure what this future code developer wants from  `AttribSubdomains::isMatch` . For example, maybe the query attrib is passed a std::set and the code developer expects  `AttribSubdomains::isMatch` to return true if any one exec flag in the query object is in the attrib object for a UO. Or maybe they want  `AttribSubdomains::isMatch`  to return true if  the querry set is a subset of the attrib UO object. 

I also don't think the assertion should trip if `_vals` is empty. If a UO had an `AttribPreAux` or `AttribPostAux` with an empty `_vals` member, it would simply mean they don't execute preaux of postaux on any exec flag.

Reply:
Your assertion works fine for the current code as it stands in moose - but within the APIs intended use constraints, it is possible (or should be possible) to write code that would trip your assertion incorrectly.  The match and equality attributes need to operate symmetrically.  If `foo._vals.size()` is one and `bar._vals.size()` is 3, `foo.isMatch(bar)` needs to return the same as `bar.isMatch(foo)`.  The assertion as currently written would assert/fail in one case, but not the other.

Reply:
isMatch is now symmetric. As we discussed elsewhere, the assertion is removed